src/axelrod_dojo/archetypes/gambler.py

```diff
@@ -55,6 +66,27 @@ def create_vector_bounds(self):
 
         return lb, ub
 
+    @staticmethod
+    def mutate_pattern(pattern, mutation_probability):
+        for i in range(len(pattern)):
```

Comment:
Better to use enumerate then `range(len`

```
for i, gene in enumerate pattern:
    pattern[i] += (2 * random.random() - 1) 
```

Note that you are currently not quite using a uniform distribution (because of the `min(pattern[i], 1)`, I'd suggest:

```
for i, gene in enumerate pattern:
    r = (2 * random.random() - 1) 
    if r >= 0:
        pattern[i] += r * (1 -  gene)
    else:
        pattern += r * gene
```

Reply:
(This is because essentially if we're going to subtract `gene` is an upper bound and if we're going to add then `1 - gene` is an upper bound.

Reply:
This sounds good.

Reply:
Actually @Nikoleta-v3 apologies I believe that is way more complicated than necessary, it's just a linear map from `[0, 1]` to `[-gene, 1-gene]` so:

```
for i, gene in enumerate(pattern):
    pattern[i] += random.random() - gene
```

Will do it:

- when `random.random()` is 1 we get `1 - gene`
- when `random.random()` is 0 we get `-gene`

Reply:
Then this is equivalent to saying that:

```
pattern[i] = random.radnom()
```
I'll go with that. Thank you!

Reply:
You're completely right! Apologies for making all that waaaay more complicated than it needed to be :+1: